base/statistics.jl

```diff
@@ -9,7 +9,7 @@ function mean(f::Callable, iterable)
     end
     count = 1
     value, state = next(iterable, state)
-    total = f(value)
```

Comment:
I think it would be better if Julia could just figure out that `zero(f(value))===false` when `f=isodd` and `isa(value,Int)`. 😈